pkg/storaged/stratis-details.jsx

```diff
@@ -263,6 +265,102 @@ export const StratisPoolDetails = ({ client, pool }) => {
+                     {
+                         visible: () => !!key_desc
+                     }),
+                PassInput("passphrase", _("Disk passphrase"),
+                          {
+                              visible: () => !!key_desc,
+                              validate: val => !val.length && _("Passphrase cannot be empty"),
+                              explanation: _("Adding a keyserver requires unlocking the pool. Please provide the existing pool passphrase.")
+                          })
+            ],
+            Action: {
+                Title: _("Add"),
+                action: function (vals, progress) {
+                    return get_tang_adv(vals.tang_url)
+                            .then(adv => confirm_tang_trust(vals.tang_url, adv, key_desc, vals.passphrase));
+                }
+            }
+        });
+    }
+
+    function confirm_tang_trust(url, adv, key_desc, passphrase) {
+        dialog_open({
+            Title: _("Verify key"),
+            Body: <TangKeyVerification url={url} adv={adv} />,
+            Action: {
+                Title: _("Trust key"),
+                action: function (vals, progress) {
+                    function bind() {
+                        return pool.BindClevis("tang", JSON.stringify({ url, adv }))
+                                .then((result, code, message) => {
+                                    if (code)
+                                        return Promise.reject(message);
+                                });
+                    }
+
+                    if (key_desc) {
+                        return client.stratis_store_passphrase(key_desc, passphrase)
+                                .then(bind)
+                                .catch(ex => {
+                                    return remove_passphrase(client, key_desc)
+                                            .then(() => Promise.reject(ex));
+                                })
+                                .then(() => {
+                                    return remove_passphrase(client, key_desc);
+                                });
+                    } else
+                        return bind();
+                }
+            }
+        });
+    }
+
+    function remove_tang() {
+        dialog_open({
+            Title: _("Remove Tang keyserver?"),
+            Body: <div>
+                <p>{ fmt_to_fragments(_("Remove $0?"), <b>{tang_url}</b>) }</p>
+                <p className="slot-warning">{ fmt_to_fragments(_("Keyserver removal may prevent unlocking $0."), <b>{pool.Name}</b>) }</p>
+            </div>,
+            Action: {
+                DangerButton: true,
+                Title: _("Remove"),
+                action: function (vals) {
+                    return pool.UnbindClevis()
+                            .then((result, code, message) => {
+                                if (code)
+                                    return Promise.reject(message);
```

Comment:
These 15 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18818-20230522-090630-61e0720a-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)